### DISCUSSION TITLE
PurgeCSS doesn't safelist properly

### DISCUSSION THREAD

**Discussion by mushketer888** [tailwindlabs/tailwindcss#2961] [Help] [answered]
Hello!

It seems like it is impossible to safelist some TailwindCSS classes properly, for example:
`safelist: ['px-6', 'py-6', 'px-16', 'py-8', 'm-4', 'm-3', '-mt-32', 'pb-8', 'h-3', 'pt-12', 'h-32', '-mt-10',
        'md:mt-4', 'ml-20', 'mt-16', 'mt-0', 'mt-8', 'ml-2', 'w-3/12', 'w-4/12', 'w-2/12', 'ml-8', 'ml-12']`
It doesn't safelist some classes like w-3/12, w-2/12 etc.

**Comment by mushketer888** [comment]
Maybe you could suggest me regexp?

**Comment by mushketer888** [comment, marked as answer]
`/w-\d{1,2}\/12/`

**Comment by ekazda** [comment]
This does not seem to work for me. I am using Tailwind 1.96 and that regex does not work as `safelist` or `whitelist`. Is anyone else having this problem when trying to safelist the w-#/12 classes? Below is my code.

```
purge: {
        content: [
            '*.php',
            '!({.cache,.git,node_modules,vendor,utils,images,graphics,admin})/**/*.php',
        ],
        options: {
            whitelist: ['/w-\d{1,2}\/12/'],
        },
    }
```

I've also tried `['/^.*w-\d{1,2}\/12.*$/']` but that doesn't work either.

**Comment by rinajhinjhina** [comment]
Facing the same issue. `safelist` does not seem to be working.
